### PR TITLE
switched to mocha, cleaned up tests and upgraded deps

### PR DESCRIPTION
**Pull request by bergos**

### CODE REVIEW

File: test/arguments.test.js [hunk 1]
```diff
@@ -1,135 +1,120 @@
...
-        sinon.match.object,
-        {
-          context: options.context,
-          variables: options.variables,
-          basePath: options.basePath
-        }
-      )).toBe(true)
+      for (const key of ['basePath', 'context', 'variable']) {
+        strictEqual(called[key], options[key])
+      }
```
Comment (tpluscode) [inline test/arguments.test.js:68]: I don't see how this is "cleaner" than using sinon.

File: test/ecmaScriptLiteral.test.js [hunk 2]
```diff
@@ -1,31 +1,30 @@
...
+    let error = null
+
+    try {
       loader({ term, dataset: rdf.dataset() })
-    }).toThrow()
+    } catch (err) {
+      error = err
+    }
+
+    strictEqual(error instanceof Error, true)
```
Comment (tpluscode) [inline test/ecmaScriptLiteral.test.js:28]: Same here? Would `chai` be a readable replacement for `jest`?

File: test/arguments.test.js (hunk 1 above)
Comment (bergos) [reply test/arguments.test.js:68]: I think that was the only place the sinon code was a little bit more readable as the plain js code. E.g. I even consulted the docs for [this line](https://github.com/zazuko/rdf-loader-code/pull/18/files#diff-9b26909759830130c077d1cdf3542f68L19), but it doesn't describe what the default stub does.

File: test/ecmaScriptLiteral.test.js (hunk 2 above)
Comment (bergos) [reply test/ecmaScriptLiteral.test.js:28]: We could use `assert`. I'm just used to do it manually, cause older versions didn't support `rejects`, which should be no longer a problem.

**Pull request merged**